app/controllers/tasks_controller.rb

```diff
@@ -1,15 +1,19 @@
 class TasksController < ApplicationController
   def index
-    tasks = API::Task
-            .where(auth_id: current_user_id)
-            .includes(:framework, :latest_submission)
-            .all
-            .sort_by! { |t| Date.parse(t.due_on) }
-
-    # Move completed tasks to the bottom
-    completed_tasks = tasks.select { |t| t.status == 'completed' }
-    @tasks = (tasks - completed_tasks) + completed_tasks
+    @tasks = API::Task
+             .where(auth_id: current_user_id, status: ['unstarted', 'in_progress'])
+             .includes(:framework, :latest_submission)
+             .all
+             .sort_by! { |t| Date.parse(t.due_on) }
   end
 
   def complete; end
+
+  def history
+    @tasks = API::Task
+             .where(auth_id: current_user_id, status: 'completed')
```

Comment:
@tekin Ah, that's one of those things that are not self-evident, maybe because I'm not familiar with JSONAPI.

So if I understand you right, the call would be `API::Task.where(status: 'completed')`, and the API will do the filtering by user within its grey box?

Reply:
Yep that’s right. There is middleware that includes the user auth ID with every API request. The API app uses that to scope responses.

Reply:
Here's the PR where we did the work to automatically include the auth id in API requests: 
https://github.com/dxw/DataSubmissionService/pull/154